python/S3_VPC_ENDPOINT_ENABLED/S3_VPC_ENDPOINT_ENABLED.py

```diff
@@ -0,0 +1,399 @@
+# Copyright 2017-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+'''
+Description
+   Check whether VPC Endpoint is enabled for each VPC to access Amazon S3.
+
+Trigger
+   Periodic (because VPC Configuration Item does not record Endpoint status)
+
+Reports on:
+   AWS::EC2::VPC
+
+Rule Parameters:
+   None
+
+Scenarios:
+  Scenario 1:
+  Given: No VPC is present while using the describe_vpcs() API call
+   Then: Return NOT_APPLICABLE
+
+  Scenario 2:
+  Given: At least one VPC is present
+    And: The S3 service is not present in the list of "ServiceName" on DescribeVpcEndpoints API
+   Then: Return NON_COMPLIANT on this VPC
+
+  Scenario 3:
+  Given: At least one VPC is present
+    And: The S3 service is present in the "ServiceName" key on DescribeVpcEndpoints API
+    And: The "State" key value is not "Available"
+   Then: Return NON_COMPLIANT on this VPC
+
+  Scenario 4:
+  Given: At least one VPC is present
+    And: The S3 service is present in the "ServiceName" key on DescribeVpcEndpoints API
+    And: The "State" key value is "Available"
+   Then: Return COMPLIANT on this VPC
+'''
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::EC2::VPC'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def get_vpcendpoints(vpc_id, event):
+    compliance_results = {}
+    ec2_client = get_client('ec2', event)
+    compliance = 'NON_COMPLIANT'
+    annotate = 'There are no Amazon S3 VPC endpoints present in '+ vpc_id+'.'
+    response = ec2_client.describe_vpc_endpoints(Filters=[

```

Comment:
the service name value is region-based. For example `com.amazonaws.us-east-1.s3`.  Since the string contains the region as well, we'd have to put in service name for each region, which would be hardcoding the value for every region's endpoint. Hence I did not put in the `service-name` as a filter

Reply:
added method `get_region_from_config_arn` which strips the region from the Config ARN in the `event` payload